wallet-new/test/unit/Test/Spec/Addresses.hs

```diff
@@ -348,6 +395,137 @@ spec = describe "Addresses" $ do
                                                slice rNumOfPages rNumPerPage fixtureAddresses
                         pure (toBeCheckedAddresses === correctAddresses)
 
+        describe "Address listing with multiple Accounts (Servant)" $ do
+            prop "page 0, per page 0" $ withMaxSuccess 20 $ do
+                monadicIO $
+                    withAddressesFixtures 4 4 $ \_ layer _ _ -> do
+                        let pp = PaginationParams (Page 0) (PerPage 0)
+                        res <- runExceptT $ runHandler' $ do
+                            Handlers.listAddresses layer (RequestParams pp)
+                        case res of
+                            Right wr | null (wrData wr) -> pure ()
+                            _ -> fail ("Got " ++ show res)
+
+            prop "it yields the correct number of results" $ withMaxSuccess 20 $ do
+                monadicIO $
+                    withAddressesFixtures 3 4 $ \_ layer _ _ -> do
+                        let pp = PaginationParams (Page 1) (PerPage 40)
+                        res <- runExceptT $ runHandler' $ do
+                            Handlers.listAddresses layer (RequestParams pp)
+                        case res of
+                            Right wr -> do
+                                -- this takes into account that there is an initial account
+                                -- and each account has an initial address (but not the initial
+                                -- account thus the -1)
```

Comment:
Yes, unfortunately if you create multiple accounts for the same wallet those _won't_ come with an extra associated address. But ideally, _if_ you can bake this inside the fixture creation, test logic should hopefully simplify 😉

Reply:
I should delete this comment.